sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -111,6 +111,14 @@ statement
         (AS? query)?                                                   #createHiveTable
     | CREATE TABLE (IF NOT EXISTS)? target=tableIdentifier
         LIKE source=tableIdentifier locationSpec?                      #createTableLike
+    | replaceTableHeader ('(' colTypeList ')')? tableProvider
```

Comment:
Are there other flavors of `REPLACE TABLE` that we need to support?

Reply:
I'm not sure that we should support all of what's already here, at least not to begin with.

I think that the main use of `REPLACE TABLE` as an atomic operation is `REPLACE TABLE ... AS SELECT`. That's because the replacement should only happen if the write succeeds and the write could easily fail for a lot of reasons. Without a write, this is just syntactic sugar for a combined drop and create.

I think the initial PR should focus on just the RTAS case. That simplifies this because it no longer needs the type list. What do you think?

Reply:
I think this should support the `USING` clause that is used to pass the provider name.

Reply:
Because of the `tableProvider` field at the end I think `USING` is still supported right? As mentioned elsewhere, this is copied from CTAS.

Reply:
Is there a test for it?